### ISSUE TITLE
Glimmer support?

### ISSUE THREAD

**Issue by NullVoxPopuli** [NullVoxPopuli/ember-resources#598]
Since all the framework dependencies in this library are actually re-exports of glimmer packages, it should be fairly straightforward (behaviorally) to support both ember and glimmer.

A tricky part may be (uncertainty) typescript support for all of ember/glimmer/glint simultaneously (maybe). Glint depends on glimmer dependencies as well, so maybe this is a non issue.

**Comment by knownasilya** [comment]
Should https://github.com/NullVoxPopuli/ember-resources/blob/main/ember-resources/src/core/class-based/manager.ts#L3 use @glimmer/destroyable?

**Comment by NullVoxPopuli** [comment]
yeah, I think all the `@ember/*` imports have glimmer counterparts like that.

The only thing I'm uncertain about is `assert` from `@ember/debug`.

I like that consuming apps remove those from production builds -- but idk if glimmer has an equivalent, and then if glimmer projects also remove those from production builds.
